framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -981,60 +1071,77 @@ protected boolean visitIntersectionSubtype(
    * @param supertype a type that might be a supertype
    * @return true if {@code subtype} is a subtype of {@code supertype}
    */
-  protected boolean visitTypevarSupertype(
+  protected boolean visitType_Typevar(
       AnnotatedTypeMirror subtype, AnnotatedTypeVariable supertype) {
     return isSubtypeCaching(subtype, supertype.getLowerBound());
   }
 
   /**
-   * A type variable is a subtype if its upper bounds is below the supertype. Note: When comparing
-   * two type variables this method and visitTypevarSupertype will combine to isValid the subtypes
+   * A type variable is a subtype if its upper bound is below the supertype. Note: When comparing
+   * two type variables, this method and visitType_Typevar will combine to isValid the subtypes
    * upper bound against the supertypes lower bound.
    */
-  protected boolean visitTypevarSubtype(
+  protected boolean visitTypevar_Type(
       AnnotatedTypeVariable subtype, AnnotatedTypeMirror supertype) {
-    AnnotatedTypeMirror upperBound = subtype.getUpperBound();
-    if (TypesUtils.isBoxedPrimitive(upperBound.getUnderlyingType())
+    AnnotatedTypeMirror subtypeUpperBound = subtype.getUpperBound();
+    if (TypesUtils.isBoxedPrimitive(subtypeUpperBound.getUnderlyingType())
         && supertype instanceof AnnotatedPrimitiveType) {
-      upperBound = supertype.atypeFactory.getUnboxedType((AnnotatedDeclaredType) upperBound);
+      subtypeUpperBound =
+          supertype.atypeFactory.getUnboxedType((AnnotatedDeclaredType) subtypeUpperBound);
     }
     if (supertype.getKind() == TypeKind.DECLARED
         && TypesUtils.getTypeElement(supertype.getUnderlyingType()).getKind()
             == ElementKind.INTERFACE) {
-      // Make sure the upper bound is no wildcard or type variable
-      while (upperBound.getKind() == TypeKind.TYPEVAR
-          || upperBound.getKind() == TypeKind.WILDCARD) {
-        if (upperBound.getKind() == TypeKind.TYPEVAR) {
-          upperBound = ((AnnotatedTypeVariable) upperBound).getUpperBound();
-        }
-        if (upperBound.getKind() == TypeKind.WILDCARD) {
-          upperBound = ((AnnotatedWildcardType) upperBound).getExtendsBound();
-        }
-      }
-      // If the supertype is an interface, only compare the primary annotations.
+      // The supertype is an interface.
+      subtypeUpperBound = getNonWildcardOrTypeVarUpperBound(subtypeUpperBound);
+      // Only compare the primary annotations.
       // The actual type argument could implement the interface and the bound of
       // the type variable must not implement the interface.
```

Comment:
What is "the type variable"?  Is it `subtype`?

Reply:
I don't understand this comment either. I deleted it.